edb/graphql/types.py

```diff
@@ -396,16 +397,26 @@ def graphql_schema(self) -> GraphQLSchema:
 
     def get_gql_name(self, name: s_name.QualName) -> str:
         module, shortname = name.module, name.name
+
+        # Adjust the shortname.
+        if shortname.startswith('__'):
+            # Use '_edb' prefix to mark derived and otherwise
+            # internal types. We opt out of '__edb' because we
+            # still rely on the first occurrence of '__' in
+            # GraphQL names to separate the module from the rest
+            # of the name in some code.
+            shortname = f'_edb{shortname}'
+        elif shortname.startswith('('):
+            # Looks like a union type, so we'll need to process individual
+            # parts of the name.
+            names = []
+            for part in shortname[1:-1].split(' | '):
+                names.append(
+                    self.get_gql_name(s_name.QualName(*part.split(':', 1))))
+            shortname = '_or_'.join(names)
```

Comment:
Should we mangle this more somehow? This seems like it could plausibly collide

Reply:
Realistically, any readable GraphQL reflected name can collide, so we're already relying on unlikelihood of collisions. Eventually, in the long term, we hope to have a way to remap entire schemas (possibly via the mechanism we intend primarily to be used for forms). Generally, once remapping is possible we don't need to care as much about collisions.